tools/release/publish_release.sh

```diff
@@ -412,61 +451,34 @@ tag() {
-        cd tools/fleetctl-npm && npm publish
-
-        issues=`gh issue list -m $target_milestone --json number | jq -r '.[] | .number'`
-        for iss in $issues; do
-            is_story=`gh issue view $iss --json labels | jq -r '.labels | .[] | .name' | grep story`
-            # close all non-stories
-            if [[ "$is_story" == "" ]]; then
-                echo "Closing #$iss"
-                gh issue close $iss
-            fi
-        done
-
-        echo "Closing milestone"
-        gh api repos/fleetdm/fleet/milestones/$target_milestone_number -f state=closed
+        if [ "$announce_only" = "false" ]; then
+            # TODO more checks to validate we are ready to publish
+            gh release edit --draft=false --latest $next_tag
+            gh workflow run dogfood-deploy.yml -f DOCKER_IMAGE=fleetdm/fleet:$next_ver
+            show_spinner 200
+            dogfood_deploy=`gh run list --workflow=dogfood-deploy.yml --status in_progress -L 1 --json url | jq -r '.[] | .url'`
+            cd tools/fleetctl-npm && npm publish
+
+            issues=`gh issue list -m $target_milestone --json number | jq -r '.[] | .number'`
+            for iss in $issues; do
+                is_story=`gh issue view $iss --json labels | jq -r '.labels | .[] | .name' | grep story`
+                # close all non-stories
+                if [[ "$is_story" == "" ]]; then
+                    echo "Closing #$iss"
+                    gh issue close $iss
+                fi
+            done
 
-        # Slack
-        slack_hook_url=https://hooks.slack.com/services
-        app_id=T019PP37ALW
-        announce_text=":cloud: :rocket: The latest version of Fleet is $target_milestone.\nMore info: https://github.com/fleetdm/fleet/releases/tag/$next_tag\nUpgrade now: https://fleetdm.com/docs/deploying/upgrading-fleet"
-        if [[ "$main_release" == "true" ]]; then
-            announce_text=":cloud: :rocket: The latest version of Fleet is $target_milestone.\nMore info: https://github.com/fleetdm/fleet/releases/tag/$next_tag\nUpgrade now: https://fleetdm.com/docs/deploying/upgrading-fleet\nRelease Article: $article_url\nLinkedIn Post: $linkedin_post_url"
+            echo "Closing milestone"
+            gh api repos/fleetdm/fleet/milestones/$target_milestone_number -f state=closed
         fi
 
-        echo $announce_text
-
-        if [ "$quiet" = "false" ]; then
-            curl -X POST -H 'Content-type: application/json' \
-                --data "{\"text\":\"$announce_text\"}" \
-                $slack_hook_url/$app_id/$SLACK_GENERAL_TOKEN
-
-            curl -X POST -H 'Content-type: application/json' \
-                --data "{\"text\":\"$announce_text\nDogfood Deployed $dogfood_deploy\"}" \
-                $slack_hook_url/$app_id/$SLACK_HELP_INFRA_TOKEN
-        fi
+        # Send general announcement in #general
+        general_announce_info
```

Comment:
since dry-run is handled inside general_announce_info i would just move it outside of this if / else block and call it once